pkg/network/tracer/tracer_test.go

```diff
@@ -1881,3 +1881,119 @@ func skipIfWindows(t *testing.T) {
 		t.Skip("test unavailable on windows")
 	}
 }
+
+const (
+	numberOfRequests   = 100
+	numberOfIterations = 200
+)
+
+// TestOpenSSLVersions setups a HTTPs python server, and makes sure we are able to capture all traffic.
+func TestOpenSSLVersions(t *testing.T) {
+	if !httpSupported(t) {
+		t.Skip("HTTPS feature not available on pre 4.14.0 kernels")
+	}
+
+	if !httpsSupported(t) {
+		t.Skip("HTTPS feature not available supported for this setup")
+	}
+
+	cfg := testConfig()
+	cfg.EnableHTTPSMonitoring = true
+	cfg.EnableHTTPMonitoring = true
+	cfg.BPFDebug = true
+	tr, err := NewTracer(cfg)
+	require.NoError(t, err)
+	initTracerState(t, tr)
+	defer tr.Stop()
+
+	addr := "127.0.0.1:8001"
+	closer, err := testutil.HTTPPythonServer(t, addr, testutil.Options{
+		EnableTLS: true,
+	})
+	require.NoError(t, err)
+	defer closer()
+
+	// Giving the tracer time to install the hooks
+	time.Sleep(time.Second)
+	client, requestFn := simpleGetRequestsGenerator(t, addr)
+	var requests []*nethttp.Request
+	for i := 0; i < numberOfRequests; i++ {
+		requests = append(requests, requestFn())
+	}
+
+	client.CloseIdleConnections()
+	assertAllRequestsExists(t, tr, requests)
+}
+
+var (
+	statusCodes = []int{nethttp.StatusOK, nethttp.StatusMultipleChoices, nethttp.StatusBadRequest, nethttp.StatusInternalServerError}
+)
+
+func simpleGetRequestsGenerator(t *testing.T, targetAddr string) (*nethttp.Client, func() *nethttp.Request) {
+	var (
+		random = rand.New(rand.NewSource(time.Now().Unix()))
+		idx    = 0
+		client = &nethttp.Client{}
+	)
+
+	return client, func() *nethttp.Request {
+		idx++
+		status := statusCodes[random.Intn(len(statusCodes))]
+		req, err := nethttp.NewRequest(nethttp.MethodGet, fmt.Sprintf("https://%s/%d/request-%d", targetAddr, status, idx), nil)
+		require.NoError(t, err)
+
+		resp, err := client.Do(req)
+		require.NoError(t, err)
+		require.Equal(t, status, resp.StatusCode)
```

Comment:
the client will use another tcp connection if you doesn't read/flush the fd, that may why closing took more time as the go http client stack would not close the connections and probably keep alive. (even if you Body.Close())
The normal/expected behaviour from the client would be `io.ReadAll(resp.Body); resp.Body.Close()`